src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2074,9 +2074,54 @@ bool ApplicationManagerImpl::ConvertSOtoMessage(
   return true;
 }
 
-bool ApplicationManagerImpl::ValidateMessageBySchema(const Message& message) {
+MessageValidationResult ApplicationManagerImpl::ValidateMessageBySchema(
+    const Message& message) {
+  LOG4CXX_AUTO_TRACE(logger_);
   smart_objects::SmartObject so;
-  return ConvertMessageToSO(message, so);
+  switch (message.protocol_version()) {
+    case ProtocolVersion::kV4:
+    case ProtocolVersion::kV3:
+    case ProtocolVersion::kV2: {
+      const bool conversion_result =
+          formatters::CFormatterJsonSDLRPCv2::fromString(
+              message.json_message(),
+              so,
+              message.function_id(),
+              message.type(),
+              message.correlation_id());
+      if (!conversion_result) {
+        return INVALID_JSON;
```

Comment:
@AByzhynar I think it would be good to add some logs for this function to see what was output result if you have few different scenarios in one function.